lib/services/database/database_service.dart

```diff
@@ -0,0 +1,256 @@
+import 'dart:async';
+
+import "package:collection/collection.dart";
+import 'package:mobileprism/services/database/album_data_entry.dart';
+import 'package:mobileprism/services/database/photo_data_entry.dart';
+import 'package:path/path.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:sqflite/sqflite.dart';
+
+const String photoDataTableName = "photo_data";
+const String albumDataTableName = "album_data";
+
+late final Database? _database;
+
+class DbNotOpenException implements Exception {}
+
+class FieldsMissingException implements Exception {}
+
+class DuplicateKeyException implements Exception {}
+
+class SqlFilter {
+  final String column;
+  final String operator;
+  final String value;
+  final String? comparator;
+
+  const SqlFilter(this.column, this.operator, this.value, {this.comparator});
+
+  @override
+  String toString() {
+    if (comparator == null) {
+      return '("$column" $operator ?)';
+    } else {
+      return ' $comparator ("$column" $operator ?)';
+    }
+  }
+}
+
+Future<Database> _initDb() async {
+  try {
+    final dbPath = (await getApplicationDocumentsDirectory()).path;
+    return openDatabase(
+      join(dbPath, 'meta_data_db.db'),
+      onCreate: (db, version) async {
+        createPhotoDataTable(db);
+        createAlbumDataTable(db);
```

Comment:
Im not sure. They also did not "await" the creation so I also did not "await" it. But i guess it would not hurt. ("https://docs.flutter.dev/cookbook/persistence/sqlite#example)